### PR TITLE
[PL-30725] Added details for GCP secret manager

### PR DESCRIPTION
**Pull request by rashminandasahoo**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md [hunk 1]
```diff
@@ -54,7 +55,10 @@ See [Supported Platforms and Technologies](../../../starthere-firstgen/supporte
 
 * **Harness Permissions:** to add a GSM secret manager, a Harness User must belong to a Harness User Group with the **Manage Secrets Managers** Account Permission.
 * **GCP Permissions:** The GCP Service Account you use in the **Google Secrets Manager Credentials File** should have the following IAM roles:
-	+ `roles/secretmanager.admin` or `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager`.
+	+ `roles/secretmanager.admin` and `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager` for Harness to be able to manage existing and new secrets.
```
Comment (KapilDigani) [inline docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:58]: for Harness to be able to manage existing secrets and to create new ones

File: docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md [hunk 2]
```diff
@@ -54,7 +55,10 @@ See [Supported Platforms and Technologies](../../../starthere-firstgen/supporte
...
+	+ `roles/secretmanager.viewer` and `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager` for Harness to be able to manage existing secrets from your GCP secret manager.
```
Comment (KapilDigani) [inline docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:59]: for Harness to be able to read secrets from your GCP secret manager

File: docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md [hunk 3]
```diff
@@ -54,7 +55,10 @@ See [Supported Platforms and Technologies](../../../starthere-firstgen/supporte
...
+* You cannot access the Google Cloud secret manager for a different project outside the service account provided. To access the secret manager in a different project, you must provide a separate service account JSON corresponding to that project.
+* Harness does not create a secret manager in Google, but simply uses the secret manager of the service account.
```
Comment (KapilDigani) [inline docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:61]: This point I'm confused about .. check with Piyush

File: docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md [hunk 4]
```diff
@@ -54,7 +55,10 @@ See [Supported Platforms and Technologies](../../../starthere-firstgen/supporte
 
 * **Harness Permissions:** to add a GSM secret manager, a Harness User must belong to a Harness User Group with the **Manage Secrets Managers** Account Permission.
 * **GCP Permissions:** The GCP Service Account you use in the **Google Secrets Manager Credentials File** should have the following IAM roles:
-	+ `roles/secretmanager.admin` or `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager`.
+	+ `roles/secretmanager.admin` and `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager` for Harness to be able to manage existing and new secrets.
+	+ `roles/secretmanager.viewer` and `roles/secretmanager.secretAccessor` and `roles/secretmanager.secretVersionManager` for Harness to be able to manage existing secrets from your GCP secret manager.
+* You cannot access the Google Cloud secret manager for a different project outside the service account provided. To access the secret manager in a different project, you must provide a separate service account JSON corresponding to that project.
```
Comment (KapilDigani) [inline docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:60]: This is confusing as this is the same behavior for every entity ... mentioning it specially here will create confusion .. let's remove it

File: docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md (hunk 3 above)
Comment (rashminandasahoo) [reply docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:61]: @piyush-harness Please see @KapilDigani's comment and confirm if this should be removed.

Comment (piyush-harness) [reply docs/first-gen/firstgen-platform/security/secrets-management/add-a-google-cloud-secrets-manager.md:61]: Yes 61 can be removed

**Review by SudheendraKatte — APPROVED** [review]

**Pull request merged**